packages/side-nav/src/vaadin-side-nav-helpers.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright (c) 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+
+/**
+ * @param {string} pathToMatch
+ * @param {string} itemPath
+ * @param {string} itemPathAliases
+ */
+export function isMatchingPath(pathToMatch, itemPath, itemPathAliases) {
```

Comment:
Let's add at least several unit tests for this function instead of testing it implicitly through `vaadin-side-nav-item`. These tests should call the function with different arguments and check the return result.

Reply:
In that case, it would maybe be better just to refactor the helper function into the side nav item class again. It will be something really basic like `sanitizePath(pagePath) === sanitizePath(itemPath)`

Reply:
Good point. Have you considered using `new URL()` for normalizing urls by the way?

```js
const base = new URL('/', document.location.href);

const pathsMatch = new URL(path1, base).pathname === new URL(path2, base).pathname;
```

Reply:
Makes perfect sense. I will refactor those parts. One thing I am not sure is if the base here will cover the removing baseURI [logic](https://github.com/vaadin/web-components/pull/5974/files#diff-f0fecca676c2b1beffcd4ef161c9ae4c96f4a8dc1a5bca1a5c58531e21c23c3dR44-R51).

Reply:
Extracted the path matching logic to `uri-utils.js` as discussed. Also adapted the tests and moved them to `uri-utils.test.js`.